Automattic/newspack-plugin
# Donations Wizard and Donations/Subscriptions handling

### All Submissions:






### Changes proposed in this Pull Request:



This PR: 
- Renames the Subscriptions Onboarding wizard to Reader Revenue Onboarding.
- Introduces revenue model selection into the Reader Revenue Onboarding wizard. The revenue model selected determines whether Subscriptions or Donations wizard is shown. (see screenshots).
- Adds a Donations wizard that will automatically create and manage the necessary WC products based on a few simple fields (see screenshot)

Screenshots in details:
<details>

Revenue model screen in Reader Revenue Onboarding:

![localhost_wp-admin_admin php_page=newspack-reader-revenue-onboarding-wizard](https://user-images.githubusercontent.com/7317227/61654379-26939980-ac71-11e9-9f98-0cc22961adec.png)

Donations wizard is shown in checklist when donations model is selected (Subscriptions is shown when subscriptions model is selected):

![localhost_wp-admin_admin php_page=newspack-checklist checklist=reader-revenue (2)](https://user-images.githubusercontent.com/7317227/61654382-27c4c680-ac71-11e9-8a03-50b71fc455d8.png)

Donations wizard is just this one simple screen:

![localhost_wp-admin_admin php_page=newspack-donations-wizard](https://user-images.githubusercontent.com/7317227/61654386-298e8a00-ac71-11e9-91cb-ca924d4f9384.png)


</details>

### How to test the changes in this Pull Request:

1. Go through Reader Revenue Onboarding wizard. Select Donations model.
2. Go to Donations wizard. Add your desired settings. Save.
3. Go to **Products > All products** in WP Admin sidebar. You should see 4 products: the grouped product containing all the donations options and monthly/yearly/once donation products. You can view the grouped product on the frontend to verify your settings saved correctly, but it needs a lot of styling love. Readers likely won't be purchasing from the product page; they will be using the donations landing page.

### Discussion

The questionnaire to determine the revenue model is somewhat different than the in-progress designs on Figma (I built it before the design was presented). I'm OK updating it if necessary; I'd like to get feedback on it first and see if it might meet our needs better than the in-progress designs. 

Tier support isn't currently implemented in the Donations wizard; I tried to keep it super straightforward, and I recommend the Subscriptions model for tier-membership-based sites. My main concern with tier support in the donation wizard is it introduces a lot of complexity both for us developing it, but more importantly for our users. I think it could be argued that once a site starts doing tiers with different benefits they are on a subscription model, since readers are paying for specific benefits. Either way, [the Reveal donation screen](https://www.revealnews.org/donate/donate-to-reveal/) can be built using just this Donations wizard (and Donations block when complete); the only difference would be it wouldn't have the auto-updating "This donation will make you Engaged." message. 

### Other information:
